pinot-core/src/test/java/org/apache/pinot/queries/ExplainPlanQueriesTest.java

```diff
@@ -0,0 +1,510 @@
+    result1.add(
+        new Object[]{"SELECT(selectList:invertedIndexCol1, invertedIndexCol2, invertedIndexCol3, jsonIndexCol1, "
+            + "noIndexCol1, noIndexCol2, noIndexCol3, rangeIndexCol1, rangeIndexCol2, rangeIndexCol3, "
+            + "sortedIndexCol1, textIndexCol1)", 2, 1});
+    result1.add(
+        new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:invertedIndexCol1, invertedIndexCol2, invertedIndexCol3, "
+            + "jsonIndexCol1, noIndexCol1, noIndexCol2, noIndexCol3, rangeIndexCol1, rangeIndexCol2, rangeIndexCol3, "
+            + "sortedIndexCol1, textIndexCol1)", 3, 2});
+    result1.add(
+        new Object[]{"PROJECT(sortedIndexCol1, noIndexCol3, rangeIndexCol1, rangeIndexCol2, jsonIndexCol1, "
+            + "invertedIndexCol1, noIndexCol2, invertedIndexCol2, noIndexCol1, invertedIndexCol3, rangeIndexCol3, "
+            + "textIndexCol1)", 4, 3});
+    result1.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 5, 4});
+    check(query1, new ResultTable(DATA_SCHEMA, result1));
+
+    String query2 = "EXPLAIN PLAN FOR SELECT 'mickey' FROM testTable";
+    List<Object[]> result2 = new ArrayList<>();
+    result2.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result2.add(new Object[]{"COMBINE_SELECT", 1, 0});
+    result2.add(new Object[]{"SELECT(selectList:'mickey')", 2, 1});
+    result2.add(new Object[]{"PROJECT(ALL)", 3, 2});
+    result2.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 4, 3});
+    check(query2, new ResultTable(DATA_SCHEMA, result2));
+
+    String query3 = "EXPLAIN PLAN FOR SELECT invertedIndexCol1, noIndexCol1 FROM testTable LIMIT 100";
+    List<Object[]> result3 = new ArrayList<>();
+    result3.add(new Object[]{"BROKER_REDUCE(limit:100)", 0, -1});
+    result3.add(new Object[]{"COMBINE_SELECT", 1, 0});
+    result3.add(new Object[]{"SELECT(selectList:invertedIndexCol1, noIndexCol1)", 2, 1});
+    result3.add(new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:invertedIndexCol1, noIndexCol1)", 3, 2});
+    result3.add(new Object[]{"PROJECT(invertedIndexCol1, noIndexCol1)", 4, 3});
+    result3.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 5, 4});
+    check(query3, new ResultTable(DATA_SCHEMA, result3));
+
+    String query4 = "EXPLAIN PLAN FOR SELECT DISTINCT invertedIndexCol1, noIndexCol1 FROM testTable LIMIT 100";
+    List<Object[]> result4 = new ArrayList<>();
+    result4.add(new Object[]{"BROKER_REDUCE(limit:100)", 0, -1});
+    result4.add(new Object[]{"COMBINE_DISTINCT", 1, 0});
+    result4.add(new Object[]{"DISTINCT(keyColumns:invertedIndexCol1, noIndexCol1)", 2, 1});
+    result4.add(new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:invertedIndexCol1, noIndexCol1)", 3, 2});
+    result4.add(new Object[]{"PROJECT(invertedIndexCol1, noIndexCol1)", 4, 3});
+    result4.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 5, 4});
+    check(query4, new ResultTable(DATA_SCHEMA, result4));
+  }
```

Comment:
I don't see tests for SELECT queries with transform functions. Also, for queries with ORDER BY

Reply:
Added test cases for these.